default/python/AI/ProductionAI.py

```diff
@@ -97,10 +97,15 @@ def cur_best_military_design_rating():
         return _best_military_design_rating_cache[current_turn]
     priority = PriorityType.PRODUCTION_MILITARY
     if priority in _design_cache:
+        # TODO: rework the below code, CombatRatingsAI and possibly  WarShipDesigner so that the below uses a rating
+        # provided by CombatRatingsAI (so as to be more suitable for its actual use and to avoid making any assumptions
+        # about how the WarShipDesigner may apply cost adjustments to its rating).
```

Comment:
Ah, yes, I had overlooked that fleet upkeep contribution, and since it can get pretty big I think I should back it out also, as part of this stopgap measure.  

I submitted this stopgap rather than trying to produce the cleaner long-term solution now, because we seem on a tight-ish timeframe for 4.7, I thought it important to get at least some improvement in for 4.7, and I am not yet back up to speed enough on ShipDesigner and CombatRatingsAI to do that myself.  But I might be able to get there quick enough.  Let's see how this planning discussion goes....

Part of my uncertainty about adding this into WarShipDesigner was just how to make the necessary information available here where we need it without having to run optimize_design() again.  It could be that optimize_design could save two ratings values for the design cache-- in addition to the current one that is adjusted for cost and other considerations, it could save a 'rated_strength' value, which for warships would be the rating a la CombatRatingsAI, for troop ships would be the effective number of troops, etc.  The places that use the design cache would need to be adjusted to unpack the extra value even if they don't use it.  How does that approach sound to you?